src/main/java/com/aws/greengrass/deployment/ShadowDeploymentListener.java

```diff
@@ -308,6 +314,24 @@ protected void shadowUpdated(Map<String, Object> desired, Map<String, Object> re
                             .log("Deployment result already reported. Ignoring shadow update at startup");
                     return;
                 }
+                // Ignore if it's the ongoing deployment. This can happen if the last shadow deployment caused restart
+                try {
+                    // Using locate instead of injection here because DeploymentService lacks usable injection
+                    // constructor. Same as in IotJobsHelper.evaluateCancellationAndCancelDeploymentIfNeeded
+                    GreengrassService deploymentServiceLocateResult =
+                            kernel.locate(DeploymentService.DEPLOYMENT_SERVICE_TOPICS);
+                    if (deploymentServiceLocateResult instanceof DeploymentService) {
+                        DeploymentTaskMetadata currentDeployment =
+                                ((DeploymentService) deploymentServiceLocateResult).getCurrentDeploymentTaskMetadata();
+                        if (currentDeployment != null && currentDeployment.getDeploymentId().equals(configurationArn)) {
+                            logger.atInfo().kv(CONFIGURATION_ARN_LOG_KEY_NAME, configurationArn)
+                                    .log("Ongoing deployment. Ignoring shadow update at startup");
+                            return;
+                        }
+                    }
+                } catch (ServiceLoadException e) {
+                    logger.atError().setCause(e).log("Failed to find deployment service");
+                }
```

Comment:
we have a dedup logic on line 336, may be update that to store lastConfigurationArn across kernel restarts by storing it as a topic?

Reply:
Persisting it across restart can avoid the duplicated deployment. But it also prevents us from retrying deployment that failed because of power cycle in the middle of a deployment.
It's still necessary to check the ongoing deployment ID to tell which case we're dealing with.